src/NHibernate/Util/TypeExtensions.cs

```diff
@@ -48,5 +53,60 @@ internal static System.Type UnwrapIfNullable(this System.Type type)
 
 			return type;
 		}
+
+		internal static IEnumerable<MemberInfo> GetFieldsOfHierarchy(this System.Type type)
+		{
+			if (!type.IsInterface)
+			{
+				var analyzing = type;
+				while (analyzing != null && analyzing != typeof(object))
+				{
+					foreach (var fieldInfo in GetUserDeclaredFields(analyzing))
+					{
+						yield return fieldInfo;
+					}
+
+					analyzing = analyzing.BaseType;
+				}
+			}
+		}
+
+		internal static IEnumerable<PropertyInfo> GetPropertiesOfHierarchy(this System.Type type)
+		{
+			if (!type.IsInterface)
```

Comment:
a) you do not need to traverse hierarchy here as `GetProperties` is capable of returning all properties of hierarchy.
b) while interface could not have fields they could have properties. So in a method above this is a simple optimization while here it prevents a valid case.

Reply:
According to the [documentation](https://docs.microsoft.com/en-us/dotnet/api/system.type.getproperties?view=netframework-4.7.2#System_Type_GetProperties_System_Reflection_BindingFlags_):

> Only protected and internal properties on base classes are returned; private properties on base classes are not returned.

The point of the change is specifically to detect private ones. Of course, private properties are not an usual thing. Usually fields are used instead. This is likely why in most cases, it is not cared for getting private properties.